clang/include/clang/Basic/AttrDocs.td

```diff
@@ -3241,6 +3241,73 @@ parameter.
   }];
 }
 
+def SYCLIntelFPGAIpInterfaceCsrDocs : Documentation {
+  let Category = DocCatFunction;
+  let Heading = "intel::ip_interface_csr";
+  let Content = [{
+``[[intel::ip_interface_csr]]`` attribute applies to a function/lambda function.
+Takes no arguments and applies to the IP component that uses CSR
+(Control-Status Registers) invocation interface. Cannot be applied multiple times
+to the same function. Cannot be combined with``[[intel::ip_interface_streaming]]``
+or ``[[intel::ip_stall_free_return]]``.
+
+if ``intel::ip_interface_csr`` is applied to a function called from a
+device kernel, the attribute is ignored and it is not propagated to the kernel.
+
+.. code-block:: c++
+
+  [[intel::ip_interface_csr]] int foo() {}
+
+  }];
+}
+
+def SYCLIntelFPGAIpInterfaceStreamingDocs : Documentation {
+  let Category = DocCatFunction;
+  let Heading = "intel::ip_interface_streaming";
+  let Content = [{
+``[[intel::ip_interface_streaming]]`` attribute applies to a function/lambda function.
```

Comment:
Similar concerns here -- I have no idea what this attribute actually *does* from this description. The last one sounded like information the backend can infer rather than requiring the user to mark a function, but with this one I'm not certain I can even guess why it's needed.